.github/workflows/ci-actions-incremental.yml

```diff
@@ -1137,7 +1137,87 @@ jobs:
           path: |
             build-reports.zip
           retention-days: 7
+      - name: Collect build JSON stats
+        shell: bash
+        run: find . -name '*runner*.json' | tar czvf build-stats.tgz -T -
+      - name: Upload build JSON stats
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-stats-${{matrix.category}}
+          path: 'build-stats.tgz'
+          retention-days: 7
 
+  native-tests-stats-upload:
+    name: Upload build stats to collector
+    if: ${{ always() && github.repository == 'quarkusio/quarkus' && endsWith(github.ref, '/main') && github.event_name != 'pull_request' && needs.native-tests.result != 'skipped' && needs.native-tests.result != 'cancelled' }}
+    needs:
+      - native-tests
+      - calculate-test-jobs
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.calculate-test-jobs.outputs.native_matrix) }}
+    runs-on: ${{matrix.os-name}}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: graalvm/mandrel
+          fetch-depth: 1
+          path: workflow-quarkus
+      - uses: actions/download-artifact@v4
+        with:
+          name: build-stats-${{matrix.category}}
+          path: .
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 17
+      - name: Extract and import build stats
+        env:
+          UPLOAD_TOKEN: ${{ secrets.UPLOAD_COLLECTOR_TOKEN }}
+          COLLECTOR_URL: https://collector.foci.life/api/v1/image-stats
+          TAG: quarkus-main-ci
+        shell: bash
+        run: |
+          cat > ./runner-info.json <<EOF
+          {
+            "test_version": "$GITHUB_REF_NAME",
+            "quarkus_version": "$GITHUB_SHA",
+            "jdk_version": "$(java -version 2>&1 | head -n 2 | tail -n 1)",
+            "operating_system": "$RUNNER_OS",
+            "architecture": "$RUNNER_ARCH",
+            "memory_size_bytes": "$(awk '/MemTotal/ {print $2 * 1024}' /proc/meminfo)",
+            "memory_available_bytes": "$(awk '/MemAvailable/ {print $2 * 1024}' /proc/meminfo)",
+            "description": "Quarkus CI github runner on $GITHUB_REF_NAME branch/tag",
+            "triggered_by": "Quarkus CI"
+          }
```

Comment:
Runner info extracted here might be different from the actual GHA runner that ran the native integration test, no? Should we not collect the info where the native tests ran?

Reply:
The only thing I expect to actually change is `memory_available_bytes` as AFAIK all free-tier github runners have the same resources. So I am not sure if it's worth creating a new runner for each IT-category (since each category runs on its own runner). No strong opinions though.

Reply:
So maybe we should keep the runner info empty (or some value indicating the values are somewhat meaningless), so as to avoid concluding something from the runner info. That's the only worry. The only info relevant for quarkus should be the `test_version` and `quarkus_version`, right?

Reply:
> So maybe we should keep the runner info empty (or some value indicating the values are somewhat meaningless), so as to avoid concluding something from the runner info.

That makes sense

> The only info relevant for quarkus should be the test_version and quarkus_version, right?

Right, that should be enough for https://github.com/quarkusio/quarkus/issues/40076